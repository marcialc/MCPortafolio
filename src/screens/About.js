import React, { Component } from "react";

export const age = (born, today) => {
  return today - born;
}

export default class About extends Component {
  render() {
    return (
      <section id="about">
        <div className="row">
          <div className="three columns">
            <img className="profile-pic" src="assets/me2.jpg" alt="" />
          </div>

          <div className="nine columns main-col">
            <h2>About Me</h2>
            <p>Currently a Senior in Florida International University (FIU) Pursing a Bachelors of Science in Computer Science. Very passionated about about my field and whats to come and the new tehcnologies and challenges it brings.</p>
            <p>I'm {age(1995, 2019)} years old</p>
            <div className="row">
              <div className="columns contact-details">
                <h2>Contact</h2>
                <p className="address">
                  <span>Marcial Cabrera</span>
                  <br></br>
                  <span>Miami, FL</span>
                  <br></br>
                  <span>marcialandres06@gmail.com</span>
                </p>
              </div>
            </div>
          </div>
        </div>
      </section>
    );
  }
}
