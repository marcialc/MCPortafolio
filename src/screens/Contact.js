import React from 'react';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import Content from '../components/Content';
import Axios from 'axios';

class Contact extends React.Component {

    /**
     * SEND GRID INFO
     * 
     * API KEY
     * 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
     * 
     * USERNAME
     * app145000746@heroku.com
     * 
     * PASSWORD
     * b7pemmda6727
     **/ 

    

    constructor(props) {
        super(props);
        this.state = {
            name: '',
            email: '',
            message: '',
            disabled: false,
            emailSent: null,
        }
    }


    handleChange = (event) => {
        const target = event.target;
        const value = target.type === 'checkbox' ? target.checked : target.value;
        const name = target.name;

        this.setState({
            [name]: value
        })
    }


    handleSubmit = (event) => {
        event.preventDefault();

        console.log(event.target);

        this.setState({
            disabled: true
        });

        Axios.post('http://localhost:3030/api/email', this.state)
            .then(res => {
                if(res.data.success) {
                    this.setState({
                        disabled: false,
                        emailSent: true
                    });
                } else {
                    this.setState({
                        disabled: false,
                        emailSent: false
                    });
                }
            })
            .catch(err => {
                console.log(err);

                this.setState({
                    disabled: false,
                    emailSent: false
                });
            })

    }


    render() {
        return(
            <div id="contact">
                <div className="contact-info">
                  <h3> Marcial Cabrera </h3>
                  <p> marcialandres06@gmail.com </p>
                  <p> (786)-712-3402 </p>
                  <p style={{margin:0, paddingTop:"10px"}}> If you have any questions shoot me an email (: </p>
                </div>
                <Content>
                    <Form onSubmit={this.handleSubmit}>
                        <Form.Group>
                            <Form.Label htmlFor="full-name">Full Name</Form.Label>
                            <Form.Control id="full-name" name="name" type="text" value={this.state.name} onChange={this.handleChange} />
                        </Form.Group>


                        <Form.Group>
                            <Form.Label htmlFor="email">Email</Form.Label>
                            <Form.Control id="email" name="email" type="email" value={this.state.email} onChange={this.handleChange} />
                        </Form.Group>


                        <Form.Group>
                            <Form.Label htmlFor="message">Message</Form.Label>
                            <Form.Control id="message" name="message" as="textarea"rows="3" value={this.state.message} onChange={this.handleChange} />
                        </Form.Group>


                        <Button className="d-inline-block" variant="primary" type="submit" disabled={this.state.disabled}>
                            Send
                        </Button>


                        {this.state.emailSent === true && <p className="d-inline success-msg"> Email Sent </p>}
                        {this.state.emailSent === false && <p className="d-inline err-msg"> Email Not Sent </p>}
                    </Form>
                </Content>
            </div>
        );
    }

}

export default Contact;
