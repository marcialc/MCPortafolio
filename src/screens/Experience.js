import React, { Component } from "react";
import Data from "../Data";

export default class Experience extends Component {
  render() {
    return (
      <section id="experience">
        {/* Education Section */}
        <div className="row education">
          <div className="three columns header-col">
          <h1>
              <span>education</span>
            </h1>
          </div>
          <div className="nine columns main-col">
            <div>
              <h2>Florida International University (FIU)</h2>
              <div className="major">
                B.S Computer Science
                <br />
                <p className="date">
                  Expected Graduation Date: Fall 2019 
                </p>
              </div>
            </div>
          </div>
        </div>

        {/* Work Section */}
        <div className="row work">
          <div className="three columns header-col">
            <h1>
              <span>work</span>
            </h1>
          </div>
          <div className="nine columns main-col">
            {Data.work &&
              Data.work.map(work => {
                return (
                  <div className="row item">
                    <div className="twelve columns">
                      <h2>
                        {work.CompanyName}{" "}
                        <em
                          style={{
                            fontSize: 12,
                            paddingLeft: 30
                          }}
                        >
                          {work.startDate} - {work.endDate}
                        </em>
                      </h2>

                      <p className="title">{work.title}</p>
                      <p className="date"> {work.description}</p>
                    </div>
                  </div>
                );
              })}
          </div>
        </div>


        {/* Projects Section */}
        <div className="row work">
          <div className="three columns header-col">
            <h1>
              <span>projects</span>
            </h1>
          </div>
          <div className="nine columns main-col">
            {Data.projects &&
              Data.projects.map(project => {
                return (
                  <div className="row item">
                    <div className="twelve columns">
                      <h2>
                        {project.title}{" "}
                        <em
                          style={{
                            fontSize: 12,
                            paddingLeft: 30
                          }}
                        >
                          {project.date}
                        </em>
                      </h2>

                      <p className="title">{project.name}</p>
                      <p className="date"> {project.description}</p>
                    </div>
                  </div>
                );
              })}
          </div>
        </div>



        {/* Programming Languages Section */}
        <div className="row skill">
          <div className="three columns header-col">
          <h1>
              <span>
                Programming
                <p />
                Languages
              </span>
            </h1>
          </div>

          <div className="nine columns main-col">
            <div className="bars">
              <ul>
                {Data.programmminglanguages &&
                  Data.programmminglanguages.map(lang => {
                    return (
                      <li>
                        <span
                          className={`bar-expand ${lang.skillname.toLowerCase()}`}
                        ></span>
                        <em>{lang.skillname}</em>
                      </li>
                    );
                  })}
              </ul>
            </div>
          </div>
        </div>
      </section>
    );
  }
}
