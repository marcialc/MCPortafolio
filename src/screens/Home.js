import React, { Component } from "react";
import Data from "../Data";

export default class Home extends Component {
  render() {
    return (
      <React.Fragment>
        <header id="home">
          <nav id="nav-wrap">           
            <ul id="nav" className="nav">
              <li className="current">
                <a className="smoothscroll" href="#home">
                  Home
                </a>
              </li>
              <li>
                <a className="smoothscroll" href="#about">
                  About
                </a>
              </li>
              <li>
                <a className="smoothscroll" href="#experience">
                  Experience
                </a>
              </li>
            </ul>
          </nav>
          <div className="row banner">
            <div className="banner-text">
              <h1 className="responsive-headline">Marcial Cabrera</h1>
              <h3>Software Engineer / Front-end Developer / Mobile Developer</h3>
              <ul className="social">
                {Data.socialLinks &&
                  Data.socialLinks.map(item => {
                    return (
                      <li key={item.name}>
                        <a href={item.url}>
                          <i className={item.className} />
                        </a>
                      </li>
                    );
                  })}
              </ul>
            </div>
          </div>
        </header>
      </React.Fragment>
    );
  }
}
