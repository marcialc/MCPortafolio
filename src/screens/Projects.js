import React from "react";

export const Projects = () => (
  <div>
    <h2>Projects Page</h2>
    <p>
      Just make little strokes like that. All you have to do is let your
      imagination go wild. Very easy to work these to death. Trees grow however
      makes them happy. Use what happens naturally, don't fight it. I spend a
      lot of time walking around in the woods and talking to trees, and
      squirrels, and little rabbits and stuff.
    </p>
  </div>
);
