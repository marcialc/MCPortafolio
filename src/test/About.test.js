import React from 'react';
import About from "../screens/About";
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {age} from '../screens/About';

configure({adapter: new Adapter()});

describe('React component test with Enzyme', () => {
  it('renders home crashing', () => {
     shallow(<About/>);
   });
});

it('renders correctly', () => {
  const tree = renderer.create(<About />).toJSON()
  expect(tree).toMatchSnapshot()
})

