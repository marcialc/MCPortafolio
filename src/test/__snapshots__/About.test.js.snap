// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <section
    id="about"
  >
    <div
      className="row"
    >
      <div
        className="three columns"
      >
        <img
          alt="me"
          className="profile-pic"
          src="assets/me2.jpg"
        />
      </div>
      <div
        className="nine columns main-col"
      >
        <h2>
          About Me
        </h2>
        <p>
          Florida International University (FIU) Alumni with a Bachelors of Science in Computer Science. Very passionated about about my field and whats to come and the new tehcnologies and challenges it brings.
        </p>
      </div>
    </div>
  </section>
  <section
    id="experience"
  >
    <div
      className="row education"
    >
      <div
        className="three columns header-col"
      >
        <h1>
          <span>
            education
          </span>
        </h1>
      </div>
      <div
        className="nine columns main-col"
      >
        <div>
          <h2>
            Florida International University (FIU)
          </h2>
          <div
            className="major"
          >
            Bachelor of Science in Computer Science
            <br />
            <p
              className="date"
            >
              Graduated: Fall 2019
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row work"
    >
      <div
        className="three columns header-col"
      >
        <h1>
          <span>
            work
          </span>
        </h1>
      </div>
      <div
        className="nine columns main-col"
      >
        <div
          className="row item"
        >
          <div
            className="twelve columns"
          >
            <h2>
              American Express
               
              <em
                style={
                  Object {
                    "fontSize": 12,
                    "paddingLeft": 30,
                  }
                }
              >
                June 2019
                 - 
                August 2019
              </em>
            </h2>
            <p
              className="title"
            >
              Mobile Engineer Intern
            </p>
            <p
              className="date"
            >
               
              Conducted code reviews and tested ongoin sprints while being a core memeber of a Agile team. Designed and developed on existing movile application on multiple channels such as React Native, Android and Java services/APIs.
            </p>
          </div>
        </div>
        <div
          className="row item"
        >
          <div
            className="twelve columns"
          >
            <h2>
              CodeArt
               
              <em
                style={
                  Object {
                    "fontSize": 12,
                    "paddingLeft": 30,
                  }
                }
              >
                June 2019
                 - 
                Present
              </em>
            </h2>
            <p
              className="title"
            >
              Programming Teacher
            </p>
            <p
              className="date"
            >
               
              As an instructor at CodeArt, my goals are to help the cause that CodeArt is continuing to support by opening the doors to the previously male-dominated field of coding, CodeArt inspires young women to explore coding, showcasing their talents, and pursue careers in this growing industry.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row projects"
    >
      <div
        className="three columns header-col"
      >
        <h1>
          <span>
            projects
          </span>
        </h1>
      </div>
      <div
        className="nine columns main-col"
      >
        <div
          className="row item"
        >
          <div
            className="twelve columns"
          >
            <h2>
              Full-Stack
               
              <em
                style={
                  Object {
                    "fontSize": 12,
                    "paddingLeft": 30,
                  }
                }
              >
                August 2019
              </em>
            </h2>
            <p
              className="title"
            >
              Portfolio
            </p>
            <p
              className="date"
            >
               
              Developed front-end using Redux and React.js, also implemented back-end using node.js, java spring and MongoDB. Jest testing framework was used for code consistency and unit testing.
            </p>
          </div>
        </div>
        <div
          className="row item"
        >
          <div
            className="twelve columns"
          >
            <h2>
              Neural Network / iOS App
               
              <em
                style={
                  Object {
                    "fontSize": 12,
                    "paddingLeft": 30,
                  }
                }
              >
                February 2019
              </em>
            </h2>
            <p
              className="title"
            >
              ParasiteAI
            </p>
            <p
              className="date"
            >
               
              For creating the neural network we used Flaks web app and it was hosted in Google Could App Engine. Created an iOS app that interacted with the neural network using 8base. It also won “best design” and third place overall.
            </p>
          </div>
        </div>
        <div
          className="row item"
        >
          <div
            className="twelve columns"
          >
            <h2>
              React-Native App
               
              <em
                style={
                  Object {
                    "fontSize": 12,
                    "paddingLeft": 30,
                  }
                }
              >
                August 2018
              </em>
            </h2>
            <p
              className="title"
            >
              Bank-Health
            </p>
            <p
              className="date"
            >
               
              Build a react native app, that told you about your bank health based on your transactions and account balance using fake data from City Bank API. Stored and saved all data in firebase. Used express framework from node.js for the back-end. It was also connected to google assistance for a hand-free experience.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row skill"
    >
      <div
        className="three columns header-col"
      >
        <h1>
          <span>
            Programming
            <p />
            Languages
          </span>
        </h1>
      </div>
      <div
        className="nine columns main-col"
      >
        <div
          className="bars"
        >
          <ul>
            <li>
              <span
                className="bar-expand java"
              />
              <em>
                JAVA
              </em>
            </li>
            <li>
              <span
                className="bar-expand javascript"
              />
              <em>
                JavaScript
              </em>
            </li>
            <li>
              <span
                className="bar-expand swift"
              />
              <em>
                Swift
              </em>
            </li>
            <li>
              <span
                className="bar-expand typescript"
              />
              <em>
                TypeScript
              </em>
            </li>
            <li>
              <span
                className="bar-expand kotlin"
              />
              <em>
                Kotlin
              </em>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </section>
</div>
`;
